---
title: "Supplementary appendix to:"
subtitle: "Estimating $R_e$ and overdispersion in secondary cases from the sequence cluster size distribution of SARS-CoV-2"
author: "Emma B. Hodcroft, Martin Wohlfender, Richard Neher, Julien Riou, Christian Althaus"
date: "`r format(Sys.Date(), '%d %B %Y')`"
fontsize: 11pt
geometry: margin=1in
output:
  pdf_document:
    number_sections: true
    toc: true
    extra_dependencies: 
      array: null
      amsmath: null
      amssymb: null
      amsthm: null
      stmaryrd: null
      hyperref: null
bibliography: R_k.bib
# csl: national-library-of-medicine-brackets-no-et-al.csl
link-citations: true
---

\newtheorem{theorem}{Theorem}
\newtheorem{lemma}{Lemma}

```{r setup, include=FALSE}

knitr::opts_knit$set(root.dir =   "C:/Users/mw22f082/Documents_MW/projects/genomic_trees/genomic_trees_v3/R_and_k_final/R_k_final_github")

knitr::opts_chunk$set(echo = TRUE)

```

```{r load_packages, include=FALSE}

library(tidyverse)
library(flextable)
library(cowplot)
library(lubridate)

```

```{r run_setup_R, include=FALSE}

source("R/setup.R", local = knitr::knit_global())

```

```{r run_main_R, include=FALSE}

source("R/main.R", local = knitr::knit_global())

```

\newpage

# Data

We used sequence data collected during the SARS-CoV-2 epidemic in Switzerland, Denmark and Germany to estimate the effective reproduction number $R_{e}$ and the dispersion parameter $k$ on a monthly basis during 2021. We downloaded all available sequences of the respective country and period from GISAID\ [@shu_gisaid_2017] and grouped them into clusters of identical sequences. A cluster is assigned to a month if there is a sequence in the cluster that has been sampled in the month. The same cluster can therefore be assigned to several months. Furthermore, we retrieved the daily numbers of new confirmed cases from the WHO Coronavirus (COVID-19) Dashboard\ [@noauthor_who_nodate] as well as the number of sequences on a biweekly basis from CoVariants\ [@hodcroft_covariants_2023]. The number of sequences was averaged to a daily basis, then summarized by month. The collected data is presented in the following tables.

## Switzerland

```{r data_overview_ch_1, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cases_sequences_clusters_ch_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:8, width = c(0.85, 0.65, 0.65, 0.65, 0.65, 1.1, 1.1, 0.65), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 1: Overview of data from Switzerland. \nNocc: Number of confirmed cases   Nos: Number of sequences   Nocic: Number of cases in clusters \nNoc: Number of clusters   Solc: Size of largest clusters.")

# print table
knitr::knit_print(x = table_2)

```

\newpage

```{r data_overview_ch_2, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_ch_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 2A: Identical sequence clusters from Switzerland.")

# print table
knitr::knit_print(x = table_2)

```

```{r data_overview_ch_3, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_ch_2021_months_percentage

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 2B: Identical sequence clusters from Switzerland (proportion of clusters).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Denmark

```{r data_overview_dk_1, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cases_sequences_clusters_dk_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:8, width = c(0.85, 0.65, 0.65, 0.65, 0.65, 1.1, 1.1, 0.65), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 3: Overview of data from Denmark \nNocc: Number of confirmed cases   Nos: Number of sequences   Nocic: Number of cases in clusters \nNoc: Number of clusters   Solc: Size of largest clusters.")

# print table
knitr::knit_print(x = table_2)

```

\newpage

```{r data_overview_dk_2, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_dk_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 4A: Identical sequence clusters from Denmark")

# print table
knitr::knit_print(x = table_2)

```

```{r data_overview_dk_3, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_dk_2021_months_percentage

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 4B: Identical sequence clusters from Denmark (proportion of clusters).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Germany

```{r data_overview_de_1, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cases_sequences_clusters_de_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:8, width = c(0.85, 0.65, 0.65, 0.65, 0.65, 1.1, 1.1, 0.65), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 5: Overview of data from Germany \nNocc: Number of confirmed cases   Nos: Number of sequences   Nocic: Number of cases in clusters \nNoc: Number of clusters   Solc: Size of largest clusters.")

# print table
knitr::knit_print(x = table_2)

```

\newpage

```{r data_overview_de_2, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_de_2021_months

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 6A: Identical sequence clusters from Germany")

# print table
knitr::knit_print(x = table_2)

```

```{r data_overview_de_3, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_cluster_sizes_intervals_de_2021_months_percentage

# define width of columns
table_1 <- width(table_0, j = 1:11, width = dim_pretty(table_0)$widths * 6.3 / sum(dim_pretty(table_0)$widths), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 6B: Identical sequence clusters from Germany (proportion of clusters).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

# Probability Theory

\begin{lemma} \label{lem_taylor} For all $R, k \in \mathbb{R}_{> 0}$, for all $\mu \in \left[ 0, 1 \right]$ and for all $j \in \mathbb{Z}_{\geq 0}$:
\begin{equation*}
\sum_{l = 0}^{\infty} \frac{ \Gamma \left( k + j + l \right)}{ l ! \Gamma \left( k + j \right)}  \left( \frac{\mu R}{k + R} \right)^{l} = \left( \frac{k + R}{k + \left(1 - \mu \right) R} \right)^{k + j}.
\end{equation*}
\end{lemma}

\begin{proof}
We define
\begin{equation*}
f_{j} \left( x \right) = x^{- \left( k + j \right)}.
\end{equation*}
By Taylor's theorem:
\begin{align*}
\left( \frac{k + R}{k + \left(1 - \mu \right) R} \right)^{k + j} &= f_{j} \left( \frac{k + \left(1 - \mu \right) R}{k + R} \right) \\
&= f_{j} \left( 1 \right) + \sum_{l = 1}^{\infty} \frac{f_{j}^{\left( l \right)} \left( 1 \right)} {l !} \left( \frac{k + \left(1 - \mu \right) R}{k + R} - 1 \right)^{l} \\
&=1 + \sum_{l = 1}^{\infty}  \frac{ \prod_{m=0}^{l-1} - \left( k + j + m \right)}{ l ! }  \left( - \frac{\mu R}{k + R} \right)^{l} \\
&=1 + \sum_{l = 1}^{\infty}  \frac{ \prod_{m=0}^{l-1} \left( k + j + m \right)}{ l ! }  \left( \frac{\mu R}{k + R} \right)^{l} \\
&= \sum_{l = 0}^{\infty} \frac{ \Gamma \left( k + j + l \right)}{ l ! \Gamma \left( k + j \right)}  \left( \frac{\mu R}{k + R} \right)^{l} \\
\end{align*}
\end{proof}

\begin{lemma} \label{lem_series_1} For all $R, k \in \mathbb{R}_{> 0}$, for all $\mu \in \left[ 0, 1 \right]$ and for all $j \in \mathbb{Z}_{\geq 0}$:
\begin{equation*}
\sum_{i = j}^{\infty} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \mu R \right)^{i - j} \left( k + \left( 1 - \mu \right) R \right)^{k + j} \left( k + R \right)^{- \left(k + i \right)}  = 1
\end{equation*}
\end{lemma}

\begin{proof}
It holds:
\begin{align*}
&\sum_{i = j}^{\infty} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \mu R \right)^{i - j} \left( k + \left( 1 - \mu \right) R \right)^{k + j} \left( k + R \right)^{- \left(k + i \right)} \\
&=\left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k} \sum_{i = j}^{\infty} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \mu R \right)^{i - j} \left( k + \left( 1 - \mu \right) R \right)^{j} \left( k + R \right)^{- i}  \\
&=\left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k} \sum_{l = 0}^{\infty} \frac{ \Gamma \left( k + j + l \right)}{ l ! \Gamma \left( k + j \right)} \left( \mu R \right)^{l} \left( k + \left( 1 - \mu \right) R \right)^{j} \left( k + R \right)^{- \left( j + l \right)} \\
&=\left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k} \sum_{l = 0}^{\infty} \frac{ \Gamma \left( k + j + l \right)}{ l ! \Gamma \left( k + j \right)}  \left( \frac{k + \left( 1 - \mu \right) R}{ k + R } \right)^{j} \left( \frac{\mu R}{k + R} \right)^{l} \\
&=\left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k + j} \sum_{l = 0}^{\infty} \frac{ \Gamma \left( k + j + l \right)}{ l ! \Gamma \left( k + j \right)}  \left( \frac{\mu R}{k + R} \right)^{l}
\end{align*}
We conclude by applying Lemma \ref{lem_taylor}.
\end{proof}

\begin{theorem} \label{thrm_neg_binom}
Let $R, k \in \mathbb{R}_{> 0}$ and let $\mu \in \left[ 0, 1 \right]$. Furthermore, let $X$ be an integer-valued random variable following a negative binomial distribution with parameters $R$ and $k$,
\begin{equation*}
\text{i.e.} \quad \mathbb{P} \left( X = i \right) = \frac{ \Gamma \left( k + i \right)}{i ! \Gamma \left( k \right)} \left( \frac{k}{k+ R} \right)^{k} \left( \frac{R}{k + R} \right)^{i} \quad \forall \, i \in \mathbb{Z}_{\geq 0}.
\end{equation*}
The discrete random variable $Y$ with
\begin{equation*}
\mathbb{P} \left( Y = j \right) = \sum_{i = j}^{\infty} \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i - j} \, \mathbb{P} \left( X = i \right) \quad \forall \, j \in \mathbb{Z}_{\geq 0}
\end{equation*}
follows a negative binomial distribution with parameters $\left( 1 - \mu \right) R$ and $k$.
\end{theorem}

\begin{proof}
For all $j \in \mathbb{Z}_{\geq 0}$ we get:
\begin{align*}
\mathbb{P} \left( Y = j \right) &= \sum_{i = j}^{\infty} \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i-j} \, \mathbb{P} \left( X = i \right) \\
&= \sum_{i = j}^{\infty}  \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{i ! \Gamma \left( k \right)} \left( \frac{k}{k+ R} \right)^{k} \left( \frac{R}{k + R} \right)^{i} \\
&= \left( \frac{k}{k + \left( 1 - \mu \right) R } \right)^{k} \sum_{i = j}^{\infty}  \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{i ! \Gamma \left( k \right)} \left( \frac{k}{k+ R} \right)^{k} \left( \frac{k + \left( 1 - \mu \right) R }{k} \right)^{k}\left( \frac{R}{k + R} \right)^{i} \\
&= \left( \frac{k}{k + \left( 1 - \mu \right) R } \right)^{k} \sum_{i = j}^{\infty}  \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{i ! \Gamma \left( k \right)} \left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k}\left( \frac{R}{k + R} \right)^{i} \\
&= \frac{\Gamma \left( k + j \right)}{j ! \Gamma \left( k \right)} \left( \frac{k}{k + \left( 1 - \mu \right) R } \right)^{k} \sum_{i = j}^{\infty}  \left( 1 - \mu \right)^{j} \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k}\left( \frac{R}{k + R} \right)^{i}
\end{align*}

Furthermore,
\begin{align*}
&\sum_{i = j}^{\infty}  \left( 1 - \mu \right)^{j} \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k}\left( \frac{R}{k + R} \right)^{i}  \\
&= \left( \frac{\left( 1 - \mu \right) R }{k + \left( 1 - \mu \right) R} \right)^{j} \sum_{i = j}^{\infty}  \mu^{i-j} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \frac{k + \left( 1 - \mu \right) R }{k + R} \right)^{k}\left( \frac{k + \left( 1 - \mu \right) R}{ R } \right)^{j} \left( \frac{R}{k + R} \right)^{i} \\
&= \left( \frac{\left( 1 - \mu \right) R }{k + \left( 1 - \mu \right) R} \right)^{j} \sum_{i = j}^{\infty} \frac{ \Gamma \left( k + i \right)}{ \left( i - j \right) ! \Gamma \left( k + j \right)} \left( \mu R \right)^{i - j} \left( k + \left( 1 - \mu \right) R \right)^{k + j} \left( k + R \right)^{- \left(k + i \right)}
\end{align*}
By Lemma \ref{lem_series_1}, we get
\begin{equation*}
\mathbb{P} \left( Y = j \right) = \frac{\Gamma \left( k + j \right)}{j ! \Gamma \left( k \right)} \left( \frac{k}{k + \left( 1 - \mu \right) R } \right)^{k} \left( \frac{\left( 1 - \mu \right) R }{k + \left( 1 - \mu \right) R} \right)^{j}
\end{equation*}
\end{proof}

\newpage

# Model of cluster size distribution

Both for inferring the effective reproduction number $R_{e}$ and the dispersion parameter $k$ from genomic sequence data and for simulating clusters of identical sequences we built upon a classical branching process setup that we applied to viral transmission. Starting with one source case, the cases of the subsequent generation are determined by a recurrence relation. Ordering and connecting the cases according to their generation and direct predecessor, we can represent the disease outbreak as a graph, more precisely as a rooted tree. This is why in this paragraph we often use the language of graph theory, denoting cases as nodes, or all cases sharing a common direct predecessor as offspring of a node.

We start with the same assumption that Blumberg and Lloyd-Smith\ [@blumberg_comparing_2013] have made: For each node, the number of offspring $T$ follows a negative binomial distribution with mean $R_{e}$ and dispersion parameter $k$,
\begin{equation*}
\mathbb{P} \left( T = i \right) = \frac{ \Gamma \left( k + i \right)}{i ! \; \Gamma \left( k \right)} \left( \frac{k}{k + R_{e}} \right)^{k} \left( \frac{R_{e}}{k + R_{e}} \right)^{i} \quad \forall \, i \in \mathbb{Z}_{\geq 0}.
\end{equation*}

\noindent Having set up the scaffolding of our transmission model, we move to integrating the mutation process of the virus to separate different viral clades. At each node the occurrence of a mutation is modelled by a random variable following a Bernoulli distribution with parameter $\mu$. We assume that at each node a mutation occurs independently of all other nodes. If a mutation happens at a node, all direct offspring of that particular node receive the mutation. The mutation probability $\mu$ is determined as

\begin{equation*}
\mu = 1 - \mathrm{e}^{- M D / 365.25},
\end{equation*}

\noindent where $M$ is the yearly mutation rate of SARS-CoV-2 and $D$ the mean generation interval.

\noindent In order to determine the distribution of the size of the identical sequence clusters, we need to determine the distribution of the number of offspring $V$ of a node that belong to the same identical sequence cluster as their parent. The number of offspring within the same identical sequence cluster of a node is the number of children of the node at which no mutation occurred, i.e. the genotype of the virus these children transmit to their offspring is identical to the genotype of the virus they received from their parent. Therefore,

\begin{equation*}
\mathbb{P} \left( V = j \right) = \sum_{i = j}^{\infty} \binom{i}{j} \left( 1 - \mu \right)^{j} \mu^{i - j} \, \mathbb{P} \left( T = i \right) \quad \forall \, j \in \mathbb{Z}_{\geq 0}.
\end{equation*}

\noindent $\mathbb{P} \left( V = j \right)$ is the probability that at $j$ children of a node no mutation occurs (and at all other children a mutation occurs), which means that these $j$ children belong to the same cluster as their parent. By Theorem\ \ref{thrm_neg_binom}, $V$ follows a negative binomial distribution with mean $R_g = \left( 1 - \mu \right) R_{e}$, the genomic reproduction number, and dispersion parameter $k$, i.e.,

\begin{equation*}
\mathbb{P} \left( V = j \right) = \frac{\Gamma \left( k + j \right)}{j ! \; \Gamma \left( k \right)} \left( \frac{k}{k + R_g } \right)^{k} \left( \frac{R_g}{k + R_g} \right)^{j} \quad \forall \, j \in \mathbb{Z}_{\geq 0}.
\end{equation*}

\noindent Following Blumberg and Llyod-Smith\ [@blumberg_comparing_2013], we determine the probability that an identical sequence cluster has size $j$ by

\begin{equation*}
\mathbb{P} \left( X = j \right) = \frac{\Gamma \left( k + j - 1 \right)}{\Gamma \left( k j \right) \Gamma \left( j + 1 \right)} \frac{\left( \frac{R_g}{k} \right)^{j - 1}}{\left( 1 + \frac{R_g}{k} \right)^{kj + j - 1}} \quad \forall \, j \in \mathbb{Z}_{\geq 1}.
\end{equation*}

\noindent As did Blumberg and Llyod-Smith\ [@blumberg_inference_2013], we model the observation of each node by a random variable following a Bernoulli distribution with parameter $\tau$, the detection probability. The observation of each node is independent of the observation of all other nodes. The probability $\tau$ is determined as

\begin{equation*}
\tau = \tau_{\text{test}} \; \tau_{\text{sequence}},
\end{equation*}

\noindent where $\tau_{\text{test}}$ is the probability that a case is confirmed by a test and $\tau_{\text{sequence}}$ is the probability that the genome of a confirmed case is sequenced. Consequently, the probability of observing $j$ cases contained in an identical sequence cluster of size $\geq j$ is given by
\begin{equation*}
\mathbb{P} \left( Y = j \right) =
\begin{cases}
\sum_{i = 1}^{\infty} \left( 1 - \tau \right)^{i} \, \mathbb{P} \left( X = i \right) & \text{if} \; j = 0, \\ \\
\sum_{i = j}^{\infty} \binom{i}{j} \tau^{j} \left( 1 - \tau \right)^{i - j} \, \mathbb{P} \left( X = i \right)& \text{if} \; j \in \mathbb{Z}_{\geq 1}.
\end{cases}
\end{equation*}

\noindent As identical sequence clusters of which no case is observed do not appear in the data sets, the distribution of $Y$ does not appropriately model the size distribution of the identical sequence clusters contained in the data. To bridge this difference, we define the random variable $Z$, the number of observed cases in an identical sequence cluster of which at least one case has been observed. It holds:

\begin{equation*}
\mathbb{P} \left( Z = j \right) = \frac{\mathbb{P} \left( Y = j \right)}{1 - \mathbb{P} \left( Y = 0 \right)} \quad \forall \, j \in \mathbb{Z}_{\geq 1}.
\end{equation*}

\newpage

# Prior distributions and fixed parameters

We set weakly informative prior distributions for the effective reproduction number $R_{e}$, the dispersion parameter $k$ and the testing probability $\tau_{\text{test}}$, and a tight prior distribution for the yearly mutation rate $M$, which was informed by external data\ [@neher_contributions_2022]. We used the same prior distributions for all countries and months. For the prior distributions of $R_{e}$, $k$ and $M$ we opted for widely used probability distributions such as the gamma or normal distribution. In addition, we incorporated expert knowledge about the proportion of new cases that are confirmed by a test into the prior distribution of $\tau_{\text{test}}$. We scaled a beta distribution, usually defined on the interval $\left[0, \; 1\right]$ or $\left]0, \; 1\right[$, to the interval $\left[0.05, \; 1\right]$. The probability density function of the resulting probability distribution, which we denote by scaled beta distribution, is given by the following formula:

\begin{equation*}
f \left( x \right) = \frac{ \left( x - 0.05 \right)^{ \alpha - 1 } \left( 1 - x \right)^{ \beta - 1 }}{ \left( 1 - 0.05 \right)^{ \alpha + \beta - 1 }} \frac{\Gamma \left( \alpha + \beta \right) }{ \Gamma \left( \alpha \right) \Gamma \left( \beta \right)} \quad \forall \, x  \in \left[ 0.05, \; 1 \right],
\end{equation*}

where $\alpha$ and $\beta$ are the two shape parameters. We included the mean generation interval as a constant value\ [@ganyani_estimating_2020] into our model and determined the probability that the viral genome of a confirmed case gets sequenced by dividing the monthly number of sequences by the monthly number of confirmed cases. Supplementary tables 7A and 7B present an overview of the prior distributions as well as the fixed parameters.

```{r table_prior_distributions, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# number of confirmed cases and number of sequences per country and month
data_cases_sequences_clusters_ch_2021_months_0 <- read_csv(file = path_data_cases_sequences_clusters_ch_2021_months)
data_cases_sequences_clusters_dk_2021_months_0 <- read_csv(file = path_data_cases_sequences_clusters_dk_2021_months)
data_cases_sequences_clusters_de_2021_months_0 <- read_csv(file = path_data_cases_sequences_clusters_de_2021_months)

# determine minimal and maximal sequencing probability
min_seq_proba <- 100 * round(min(data_cases_sequences_clusters_ch_2021_months_0$`Nos/Nocc`, 
                                 data_cases_sequences_clusters_dk_2021_months_0$`Nos/Nocc`, 
                                 data_cases_sequences_clusters_de_2021_months_0$`Nos/Nocc`), 4)

max_seq_proba <- 100 * round(max(data_cases_sequences_clusters_ch_2021_months_0$`Nos/Nocc`, 
                                 data_cases_sequences_clusters_dk_2021_months_0$`Nos/Nocc`, 
                                 data_cases_sequences_clusters_de_2021_months_0$`Nos/Nocc`), 4)

data_prior_distributions <- data.frame(Symbol = c("R_{e}", "k", "M", "\\tau_{\\text{test}}"),
                                       Comment = c("Effective reproduction number", "Dispersion parameter",
                                                   "Yearly mutation rate", "Testing probability"),
                                       Support = c("\\left] 0, \\infty \\right[", "\\left] 0, \\infty \\right[",
                                                   "\\left] -\\infty, \\infty \\right[", "\\left[ 0.05, 1 \\right]"),
                                       Prior = c("Gamma(shape = 10, rate = 10)", "Gamma(shape = 5, rate = 10)",
                                                 "Normal(mean = 14, variance = 0.25)", "ScaledBeta(shape = 1, shape = 3)"))

table_data_prior_distributions <- flextable(data_prior_distributions) |>
  compose(j = "Symbol", value = as_paragraph(as_equation(Symbol))) |>
  compose(j = "Support", value = as_paragraph(as_equation(Support)))

table_data_prior_distributions <- theme_vanilla(table_data_prior_distributions)

# get table
table_0 <- table_data_prior_distributions

# define width of columns
table_1 <-  width(table_0, j = 1:4, width = c(0.7, 2.2, 0.8, 2.6), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 7A: Summary of parameters of prior distributions.")

# print table
knitr::knit_print(x = table_2)

```

```{r table_fixed_parameters, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

data_fixed_parameters <- data.frame(Symbol = c("D", "\\tau_{\\text{sequence}}"),
                                    Comment = c("Mean generation interval", "Sequencing probability"),
                                    Value = c("5.2", paste0(min_seq_proba, " \\, - \\, ", max_seq_proba, " \\%")),
                                    Unit = c("days", "percentage"),
                                    Source = c("Ganyani et al. 2020", "see Column Nos/Nocc of \nsupplementary tables 1, 3 and 5"))

table_data_fixed_parameters <- flextable(data_fixed_parameters) |>
  compose(j = "Symbol", value = as_paragraph(as_equation(Symbol))) |>
  compose(j = "Value", value = as_paragraph(as_equation(Value)))

table_data_fixed_parameters <- theme_vanilla(table_data_fixed_parameters)

# get table
table_0 <- table_data_fixed_parameters

# define width of columns
table_1 <- width(table_0, j = 1:5, width = c(0.7, 1.3, 1.2, 0.9, 2.2), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 7B: Summary of fixed parameters.")

# print table
knitr::knit_print(x = table_2)

```

```{r prior_distribution_r_k_mut_test, echo=FALSE, message=FALSE, warning=FALSE, fig.width=unit(6.3,"in")}

plot_grid_prior_distributions_paper <- add_sub(plot_grid_prior_distributions, 
                                               "Supplementary figure 1: Prior distributions of the effective reproduction number (A), \nthe dispersion parameter (B), the yearly mutation rate (C) and the testing probability (D).", 
                                               x = 0.01, 
                                               hjust = 0, 
                                               size = 11)

plot_grid_prior_distributions_paper <- ggdraw(plot_grid_prior_distributions_paper)

plot_grid_prior_distributions_paper

```

\newpage

# Additional results

## Switzerland

```{r results_model_one_overview_ch, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_one_ch_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 8A: Results of parameter estimations (Switzerland).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_one_ch_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 8B: Results of parameter estimations (Switzerland).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Denmark

```{r results_model_one_overview_dk, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_one_dk_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 9A: Results of parameter estimations (Denmark).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_one_dk_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 9B: Results of parameter estimations (Denmark).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Germany

```{r results_model_one_overview_de, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_one_de_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 10A: Results of parameter estimations (Germany).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_one_de_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 10B: Results of parameter estimations (Germany).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

# Sensitivity analysis

As an alternative to using a prior distribution that incorporates uncertainty for the testing probability in our model, we used estimates of the ascertainment rate from\ [@eales_dynamics_2023] as constant values. We set the testing probability for the months of January to August to $57.6 \, \%$ and for the months of September to December to $35 \, \%$. The following figures and tables present the results.

```{r results_models_one_two_plot, echo=FALSE, include=FALSE, message=FALSE, warning=FALSE, fig.width=6.3, fig.height=6.3}

plot_1 <- plot_results_models_one_two_R_e_variants_all_countries_2 +
  guides(color = "none",
         fill = "none")

plot_2 <- plot_results_models_one_two_k_variants_all_countries +
  guides(color = "none",
         fill = "none")

legend <- get_plot_component(plot = plot_results_models_one_two_R_e_variants_all_countries_2,
                             pattern = 'guide-box', return_all = TRUE)[[3]]

plot_1_2 <- plot_grid(plotlist = list(plot_1, plot_2, legend), ncol = 1, rel_heights = c(1, 1, 0.3))

plot_1_2_with_caption <- add_sub(plot_1_2, "Supplementary figure 2: Comparison of the estimated effective reproduction numbers \nand dispersion parameters by country and month.", x = 0.01, hjust = 0, size = 11)

plot_1_2_with_caption <- ggdraw(plot_1_2_with_caption)

```

```{r results_models_one_two_plot_a, echo=FALSE, message=FALSE, warning=FALSE, fig.width=6.3, fig.height=6.3}

plot_1_2_with_caption

```

\newpage

## Switzerland

```{r results_model_two_overview_ch, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_two_ch_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 11A: \nResults of parameter estimations with constant testing probability (Switzerland).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_two_ch_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 11B: \nResults of parameter estimations with constant testing probability (Switzerland).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Denmark

```{r results_model_two_overview_dk, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_two_dk_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 12A: \nResults of parameter estimations with constant testing probability (Denmark).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_two_dk_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 12B: \nResults of parameter estimations with constant testing probability (Denmark).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

## Germany

```{r results_model_two_overview_de, echo=FALSE, message=FALSE, warning=FALSE, ft.arraystretch=1.3}

# get table
table_0 <- table_results_model_two_de_2021_months_1

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 13A: \nResults of parameter estimations with constant testing probability (Germany).")

# print table
knitr::knit_print(x = table_2)


# get table
table_0 <- table_results_model_two_de_2021_months_2

# define width of columns
table_1 <- width(table_0, j = 1:4, width = c(0.9, 1.8, 1.8, 1.8), unit = "in")

# add footer line
table_2 <- add_footer_lines(table_1, "Supplementary table 13B: \nResults of parameter estimations with constant testing probability (Germany).")

# print table
knitr::knit_print(x = table_2)

```

\newpage

# Posterior predictive check

We ran a posterior predictive check to assess whether our simulation of identical sequence clusters is compatible with the cluster data from Switzerland, Denmark and Germany. For each country and month we randomly chose 50 samples each from the posterior distributions of the effective reproduction number $R_e$, the dispersion parameter $k$, the mutation probability $\mu$ and the detection probability $\tau$. We then simulated the same number of clusters observed in the respective country and month, running the simulation once for each of the 50 parameter combinations. We summarized the results of the simulations by the $95 \, \%$ prediction interval, calculated individually for each cluster size. The following figures present the outcome of the posterior predictive check.

## Switzerland

```{r posterior_predictive_check_ch, echo=FALSE, message=FALSE, warning=FALSE, fig.width=6.3, fig.height=6.3}

plot_comparison_clusters_ch_quantile_facet <- add_sub(plot_comparison_clusters_ch_quantile_facet, "Supplementary figure 3: Posterior predictive check for simulation of identical \nsequence clusters (Switzerland).", x = 0.01, hjust = 0, size = 11)

plot_comparison_clusters_ch_quantile_facet <- ggdraw(plot_comparison_clusters_ch_quantile_facet)

plot_comparison_clusters_ch_quantile_facet

```

\newpage

## Denmark

```{r posterior_predictive_check_dk, echo=FALSE, message=FALSE, warning=FALSE, fig.width=6.3, fig.height=6.3}

plot_comparison_clusters_dk_quantile_facet <- add_sub(plot_comparison_clusters_dk_quantile_facet, "Supplementary figure 4: Posterior predictive check for simulation of identical \nsequence clusters (Denmark).", x = 0.01, hjust = 0, size = 11)
plot_comparison_clusters_dk_quantile_facet <- ggdraw(plot_comparison_clusters_dk_quantile_facet)

plot_comparison_clusters_dk_quantile_facet

```

\newpage

## Germany

```{r posterior_predictive_check_de, echo=FALSE, message=FALSE, warning=FALSE, fig.width=6.3, fig.height=6.3}

plot_comparison_clusters_de_quantile_facet <- add_sub(plot_comparison_clusters_de_quantile_facet, "Supplementary figure 5: Posterior predictive check for simulation of identical \nsequence clusters (Germany).", x = 0.01, hjust = 0, size = 11)
plot_comparison_clusters_de_quantile_facet <- ggdraw(plot_comparison_clusters_de_quantile_facet)

plot_comparison_clusters_de_quantile_facet

```

\newpage

# Remarks

## Bibliography

Check style.

## Varia

Include code availability.

Include 3 files from GISAID.

Include figures for estimate of testing probability (?)

CHECK: Total number of clusters and sequences (sum of monthly numbers (clusters assigned to multiple months counted more than once), change to total clusters)

\newpage

# Bibliography
\noindent


